### PR TITLE
Finish google translate API. Update PDF Crowd initializer code

### PR DESCRIPTION
**Pull request by schroerbrian**
This may require some changes to our Google Cloud repo as well -- I've read conflicting things about Google Cloud Service Accounts. Something I read indicated that a Google Service Account is automatically included within the ENV of a kubernetes deployment run in Google Cloud. Note: the Google Translate package checks for several standardly named ENV vars, one of which is `TRANSLATE_CREDENTIALS`, which is used here

### CODE REVIEW

**Review by richardxia — APPROVED** [review]
This LGTM, just have a few minor Ruby style comments to make. Good work on this project!

File: Gemfile.lock [hunk 1]
```diff
@@ -116,144 +126,203 @@ GEM
...
-      actiontext (= 6.1.4.1)
-      actionview (= 6.1.4.1)
-      activejob (= 6.1.4.1)
-      activemodel (= 6.1.4.1)
-      activerecord (= 6.1.4.1)
-      activestorage (= 6.1.4.1)
-      activesupport (= 6.1.4.1)
+    rack-test (2.0.2)
+      rack (>= 1.3)
+    rails (7.0.4.2)
```
[the remark this answers is not in the document]
Comment (schroerbrian) [reply Gemfile.lock]: Oof thanks for catching that. That upgrade was accidental and was causing havoc in my local environment. I actually copied the Master `Gemfile.lock` over manually and then re-installed the `google-translate` gem and commited the update. That fixed my env, but I guess it didn't register in our Git history. Reverting the file just now did, however, as I see the committed file is on the rails 6 version. Good catch!

File: app/controllers/services_controller.rb [hunk 2]
```diff
@@ -54,21 +54,50 @@ def show
...
+        parent: "projects/#{CloudTranslation.project_id}"
+      }
+    )
+  end
+
+  def translate_html
+    response = CloudTranslation.translation_service.translate_text new_google_translate_request
+    response.translations[0].translated_text
+  rescue StandardError => e
+    if e.instance_of? Google::Cloud::ResourceExhaustedError
```
[the remark this answers is not in the document]
Comment (schroerbrian) [reply app/controllers/services_controller.rb]: Great suggestions, I have implemented this and confirmed that the error handling works.

File: app/controllers/services_controller.rb [hunk 3]
```diff
@@ -54,21 +54,50 @@ def show
...
     end
-  rescue Pdfcrowd::Error => e
-    puts "Failed to convert HTML to PDF: #{e}"
-    render plain: "There was an error getting the PDF. Please try again", status: 500
+
+    send_data PdfCrowdClient.client.convertString(html_input),
+              { type: "application/pdf",
+                disposition: "attachment; filename*=UTF-8''#{ERB::Util.url_encode('translation.pdf')} }" }
+  rescue StandardError => e
+    Rails.logger.error(e)
```
[the remark this answers is not in the document]
Comment (schroerbrian) [reply app/controllers/services_controller.rb]: Fixed! Great, I see we are getting these errors sent to Sentry ,even locally now. We really should config our Sentry system to distinguish by environment....  🤔

**Comment by schroerbrian** [comment]
> This LGTM, just have a few minor Ruby style comments to make. Good work on this project!

Thanks for the quick and helpful feedback!

File: app/controllers/services_controller.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (richardxia) [reply app/controllers/services_controller.rb]: > Fixed! Great, I see we are getting these errors sent to Sentry ,even locally now. We really should config our Sentry system to distinguish by environment.... 🤔

So, we've had a ticket open to fix this for a long time, https://github.com/ShelterTechSF/askdarcel-api/issues/402, but it looks like we putatively fixed it in https://github.com/ShelterTechSF/askdarcel-api/pull/432. Maybe we just need to set up filters on the Sentry side to ignore non-production errors?

**Pull request merged**